services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,42 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (event == NULL) return ERR_CODE_INVALID_ARG;
+  if (!thermalMgrQueueHandle) return ERR_CODE_UNKNOWN;
+  if (xQueueSend(thermalMgrQueueHandle, (const void *)event, 0) != pdTRUE) return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t checkOverUnder;
+  checkOverUnder.type = THERMAL_MGR_EVENT_OVER_UNDER;
+
+  /* Add interrupt event to the queue */
+  thermalMgrSendEvent(&checkOverUnder);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *)pvParameters;
+
   while (1) {
-    
+    float currTemp;
+    thermal_mgr_event_t nextEvent;
+
+    /* Receive next event */
+    if (xQueueReceive(thermalMgrQueueHandle, &nextEvent, (TickType_t) portTICK_PERIOD_MS * 5000) != pdTRUE) 
+      continue;
+
+    /* Check for interrupt condition else check for telemetry request */
+    if (nextEvent.type == THERMAL_MGR_EVENT_OVER_UNDER) {
+      if (currTemp > config.hysteresisThresholdCelsius) overTemperatureDetected();
```

Comment:
you should be reading the temperature to update this value once you get the interrupt to decide if its  overTemp or safe since if you don't this value will be an old value